packages/react-router/src/route.ts

```diff
@@ -291,18 +284,16 @@ type LdJsonValue = LdJsonPrimitive | LdJsonObject | LdJsonArray
 
 export type RouteLinkEntry = {}
 
-// The parse type here allows a zod schema to be passed directly to the validator
-export type SearchSchemaValidator<TInput, TReturn> =
-  | SearchSchemaValidatorObj<TInput, TReturn>
-  | SearchSchemaValidatorFn<TInput, TReturn>
-
-export type SearchSchemaValidatorObj<TInput, TReturn> = {
-  parse?: SearchSchemaValidatorFn<TInput, TReturn>
+export interface ZodSearchValidator {
+  _input: any
```

Comment:
This is the trick. We can infer any schema which has an `_input` property

Reply:
how stable is this? what if zod renames this at some point?

Reply:
this won't work for schemas using `catch`: https://www.typescriptlang.org/play/?ssl=14&ssc=1&pln=1&pc=1#code/JYWwDg9gTgLgBAbwF4F84DMoRHA5EiAE1wG4AoMgYwgDsBneOygCwFMQBDAdWBmYBFW6DgFcANvAC8cJADoIAIwBWrSjAAUCdBAgAuGbIZRgNAObqAlLMJDRE9bgUcouCygvkyMAJ5hWcAGUARjhpHz8IdDgmNk4ePkFhcRgvX38ASRDpYIBtXAB9EzARGFwAXTIAekq4WoA9AH4KanpGFnZuXmYAYQ4YFlCDRRU1TW09AyMTcytKPpYHJxc3Dwpw-wCAJkH1yOj2uK7e-uZUvzh07ezNvMKaYtKK6tq4RrIgA

Reply:
This seems by design of `zod` :(

Reply:
I know, but what I mean by this: it maybe is not as simple as it seems. I use `catch` a lot, so I could not use this (directly).
Maybe we need much more complicated type logic to fully support zod's "quirks".

Reply:
That might get ugly! D:

Reply:
see e.g. https://tanstack.com/router/latest/docs/framework/react/guide/search-params

> In the above example, we used Zod's .catch() modifier instead of .default() to avoid showing an error to the user because we firmly believe that if a search parameter is malformed, you probably don't want to halt the user's experience through the app to show a big fat error message. That said, there may be times that you do want to show an error message. In that case, you can use .default() instead of .catch().

Reply:
This is annoying because `valibot` actually behaves how we expect here for `Input` with their `fallback` variation. `Input` is the correct thing to use here but its not how we expect it to work

Reply:
@colinhacks, what would your advice be?